src/scripts/dice.ts

```diff
@@ -200,23 +200,52 @@ function combineTerms(formula: string): string {
 
     const fixedFormula = formula.replace(/^\s*-\s+/, "-").replace(/\s*\+\s*-\s*/g, " - ");
     const roll = new Roll(fixedFormula);
-    if (!roll.terms.every((t) => t.expression === " + " || t instanceof Die || t instanceof NumericTerm)) {
+    if (
+        !roll.terms.every((t) => [" - ", " + "].includes(t.expression) || t instanceof Die || t instanceof NumericTerm)
+    ) {
         // This isn't a simple summing of dice: return the roll without further changes
         return fixedFormula;
     }
 
-    const dice = roll.terms.filter((term): term is Die => term instanceof Die);
-    const diceByFaces = dice.reduce((counts: Record<number, number>, die) => {
-        counts[die.faces] = (counts[die.faces] ?? 0) + die.number;
-        return counts;
-    }, {});
-    const stringTerms = [4, 6, 8, 10, 12, 20].reduce((terms: string[], faces) => {
-        return typeof diceByFaces[faces] === "number" ? [...terms, `${diceByFaces[faces]}d${faces}`] : terms;
-    }, []);
-    const numericTerms = roll.terms.filter((term): term is NumericTerm => term instanceof NumericTerm);
-    const constant = numericTerms.reduce((runningTotal, term) => runningTotal + term.number, 0);
-
-    return new Roll([...stringTerms, constant].filter((term) => term !== 0).join("+")).formula;
+    // Parse from right to left so that when we hit an operator, we already have the term.
+    const convertedTerms = roll.terms.reduceRight((result, term) => {
+        // Ignore + terms, we assume + by default
+        if (term.expression === " + ") return result;
+
+        // - terms modify the last term we parsed
+        if (term.expression === " - ") {
+            const termToModify = result[0];
+            if (termToModify) {
+                if (termToModify.modifier) termToModify.modifier *= -1;
+                if (termToModify.diceNumber) termToModify.diceNumber *= -1;
+            }
+            return result;
+        }
+
+        result.unshift({
+            modifier: term instanceof NumericTerm ? term.number : 0,
+            diceFaces: term instanceof Die ? term.faces : 0,
+            diceNumber: term instanceof Die ? term.number : 0,
```

Comment:
How about putting this in its own object instead of inventing the 10th new way to express dice number and faces? ;)

Reply:
The entire function is temporary. To even work on afflictions, I'll need to standardize this and melee items to use the formula.ts pipeline. That's a major refactor though, and it'd be nice to get your elite/weak thing in sooner rather than later.